app/views/admin/banners/_form.html.erb

```diff
@@ -54,7 +54,22 @@
     <div class="small-12 column">
       <%= f.label :sections, t("admin.banners.banner.sections_label") %>
       <%= f.collection_check_boxes(:web_section_ids, @banner_sections, :id, :name) do |b| %>
-        <%= b.check_box %> <%= t("admin.banners.banner.sections.#{b.text}") %><br>
+        <%= b.check_box %> <%= t("admin.banners.banner.sections.#{b.text}") %><br> 
+		<% if b.text == 'budgets'%>
+		  <% Budget::Group.all.each do |group| %>
+		    <% if group.budget.phase != 'finished' %>
+		      <span style="display:inline-block; width: 20px;"></span><%= group.name %><br>
+			  <% group.headings.each do |heading| %>
+		        <span style="display:inline-block; width: 40px;"></span>
+	  		    <% if (@banner.headings!=nil) && (@banner.headings.include? heading.id.to_s) %>	  
+	              <%= check_box_tag "banner_headings[]",heading.id, checked = true, id: "banner_headings_#{heading.id}" %> <%= heading.name %><br>
```

Comment:
Hi @papayalabs. Using `serialize :headings` is a very interesting idea. However, IMHO its current implementation is making the code a bit too complicated. Ideally the view would be simple and the `banner_headings` parameters would be handled automatically with something like:

```erb
<%= f.collection_check_boxes(:budget_headings) # Probably more parameters are needed in this call %>
```
If something like this is possible using `serialize`, go for it. If not, it'd probably be better to have a `has_many :budget_headings, through: :budget_sections` relationship in the `Banner` model.